### PR TITLE
Allow leading minus or plus sign before NaN and Inf

### PR DESCRIPTION
**Pull request by nalimilan**
Fixes https://github.com/JuliaData/CSV.jl/issues/168.

### CODE REVIEW

File: src/float.jl [hunk 1]
```diff
@@ -133,7 +138,7 @@ function parsefield(io::IO, ::Type{T}, opt::CSV.Options, row, col, state, ifnull
             (!(b == LITTLEN || b == BIGN) || eof(io)) && (reset(io); b = readbyte(io); @goto checknullend)
             b = readbyte(io)
             !(b == LITTLEF || b == BIGF) && (reset(io); b = readbyte(io); @goto checknullend)
-            result = T(Inf)
+            result = Inf
```
Comment (quinnj) [inline src/float.jl]: I think we still want these to be `T(Inf)`, right?

Comment (nalimilan) [reply src/float.jl]: Woops, right, these are remnants of debugging tests. I've reverted them.

**Comment by quinnj** [comment]
Thanks for this @nalimilan! I'm digging in to the 0.7 test failures locally.

**Comment by nalimilan** [comment]
Thanks. FWIW, I've just tagged a CategoricalArrays release which passes tests on 0.7, if that can help.

**Pull request merged**